### PR TITLE
[connectors] - fix: improve Google Drive webhook response and logging

### LINKED ISSUE #5399: Throw 20x instead of 40x when deleted GDrive connectors receives webhook
When a google drive connector gets deleted we still receive the webhooks for 1-2h -> we should throw 20x instead of 40x

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

 Change response to 200 OK for non-existent connectors when receiving GDrive webhooks

[Slack thread](https://dust4ai.slack.com/archives/C05F84CFP0E/p1717313769498709)

### CODE REVIEW

File: connectors/src/api/webhooks/webhook_google_drive.ts [hunk 1]
```diff
@@ -56,21 +56,21 @@ const _webhookGoogleDriveAPIHandler = async (
   }
   const connector = await ConnectorResource.fetchById(connectorId);
   if (!connector) {
```
Comment (JulesBelveze) [inline connectors/src/api/webhooks/webhook_google_drive.ts:58]: I guess in this state we can only have a webhook corresponding to a deleted connector. Am I right, here?

**Comment by spolu** [comment]
Beware I remember there is 2 code paths for this handler related to a refactor. One is not used anymore and we should clean it up first. @lasryaric can help identify the dead code

**Comment by spolu** [comment]
Talking about this one: https://github.com/dust-tt/dust/pull/5428/files#diff-0bd684b4020526741072c412ec9e31b1bc751590e63ca78bfdb71185d929a95aR39-R47

I think it will be hard to properly fix this one without removing the dead code here?

**Comment by spolu** [comment]
Actually I think your fix works but would still be nice to remove the dead code :pray: :)

File: connectors/src/api/webhooks/webhook_google_drive.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/api/webhooks/webhook_google_drive.ts:58]: I agree with your analysis :+1:

**Review by lasryaric — APPROVED** [review]
LGTM.

**Comment by lasryaric** [comment]
> Beware I remember there is 2 code paths for this handler related to a refactor. One is not used anymore and we should clean it up first. @lasryaric can help identify the dead code

@spolu which other code path are you referring to?

**Comment by spolu** [comment]
@lasryaric this one: https://github.com/dust-tt/dust/pull/5428/files#diff-0bd684b4020526741072c412ec9e31b1bc751590e63ca78bfdb71185d929a95aR17-R47

**Comment by spolu** [comment]
One pulls the connectorId from a retrieval of a webhook from the channelId. The other retrieves the connectorId from a param. What is used in prod?

**Comment by lasryaric** [comment]
Getting the code from `req.params.connector_id` is the dead code. @JulesBelveze you can change this API handler to only retrieve the connector id via the webhook header used here[.](https://github.com/dust-tt/dust/pull/5428/files#diff-0bd684b4020526741072c412ec9e31b1bc751590e63ca78bfdb71185d929a95aR17)

**Comment by JulesBelveze** [comment]
@lasryaric Just removed the dead code part, mind double checking?

**Review by lasryaric — APPROVED** [review]
LGTM.

**Review by spolu — APPROVED** [review]
\o/

**Pull request merged**